Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -195,6 +196,35 @@ extension BrowserViewController: WKNavigationDelegate {
       return (.cancel, preferences)
     }
     
+    // Handling calendar .ics files
+    if url.pathExtension.lowercased() == "ics" {
+      let localURL = FileManager.default.temporaryDirectory.appendingPathComponent(url.lastPathComponent)
+      
+      do {
+        let (tempLocalUrl, _) = try await URLSession.shared.download(from: url)
+        if FileManager.default.fileExists(atPath: localURL.path) {
+          try FileManager.default.removeItem(at: localURL)
+        }
+        try FileManager.default.copyItem(at: tempLocalUrl, to: localURL)
```

Comment:
Is this logic is for temporarily hold ics file and not download it? Can you comment here why we have to do this ?

Reply:
You are right. This code is not needed anymore.
I used it in the older implementation where it used a document preview controller but this did not work well